### ISSUE TITLE
Request feat(controller): add disk i/o limits

### ISSUE THREAD

**Issue by aledbf** [deis/deis#3206]
Currently is possible to do something like this:

```
docker run -it --rm --name block-device-test gliderlabs/alpine:3.1 /bin/sh

wget http://www.iozone.org/src/current/iozone3_430.tar

apk-install build-base

tar xpvf iozone3_430.tar
cd iozone3_430/src/current
make linux-AMD64

./iozone -l 32 -O -i 0 -i 1 -i 2 -e -+n -r 4K -s 1G
```

Per se this is not wrong but it will consume all the disk i/o available and start a chain of events that it could put the node in a unusable/inaccessible state for the duration of the task.
Currently in docker/libcontainer there is some work to add support to limit the i/o but is not yet available (using the native driver) as a docker flag:

https://github.com/docker/libcontainer/pull/354
https://github.com/docker/libcontainer/pull/337

**Comment by bacongobbler** [comment]
@aledbf could you please explain how this could be exploited from an attacker's perspective? Not sure how this affects the controller per se.

**Comment by aledbf** [comment]
@bacongobbler forget an attacker, just a programming error in any user application can cause this.

For example (a real one from a new node.js developer in the dev cluster): run a query from a web app that returns millions of records (from mongodb) that generates a csv from the results triggered this.
Errors in the code: 
- run this type of thing in each request
- lack of knowledge of query stream in mongo
- lack of knowledge of file streams and pipes

This error that could be "harmless" in a normal use case was magnified because of a "stress" testing of the app using wrk.
During this situation the i/o wait of the node was around 80% and provoked a chain of errors not related with the app that I did not imagined :) (I could not even access the node)

This revealed that by default docker doesn't provide (by default) a mechanism to protect the system from resource starvation.

**Comment by carmstrong** [comment]
> Currently in docker/libcontainer there is some work to add support to limit the i/o but is not yet available (using the native driver) as a docker flag:

I think that's how we'd implement this, as a `deis limits:set` command for IO, which would be passed directly to the Docker daemon. However, until this is implemented in Docker itself, I don't think it's something we can do.

**Comment by aledbf** [comment]
@carmstrong this is already possible if we use lxc (the default option is native-0.2) 
```
--lxc-conf=[]              (lxc exec-driver only) Add custom lxc options --lxc-conf="lxc.cgroup.cpuset.cpus = 0,1"
```

Just in case I'm not suggesting switch to lxc.

**Comment by carmstrong** [comment]
@aledbf Is this in Docker 1.5?

**Comment by aledbf** [comment]
@carmstrong no

**Comment by aledbf** [comment]
@carmstrong released in 1.7. Now just to wait for 1.7 in CoreOS

**Comment by bacongobbler** [comment]
This is not something we'll be able to get to for the LTS release (#4776), and Deis v2 uses kubernetes so we want to rely on that to provide this feature in their manifests before adding this feature to Deis.
